### DISCUSSION TITLE
zkEVM??

### DISCUSSION THREAD

**Discussion by LandLadyFarmer** [sunflower-land/sunflower-land#2508] [General]
![image](https://user-images.githubusercontent.com/126206566/235294023-9bdcf634-0a25-4fdf-a5f3-5153f52d5174.png)

As we all know, we are affected by increasingly higher gas fees for longer periods and it is clear that most people are quite sensitive to these fees and especially for a project like this with a low entry threshold.. I don't really want to suggest but rather a wonder that has become more and more clear from all the AMAs I've listened to regarding zkEVM as a polygon recently..

 zkEvm is better suited for projects that have a higher amount of data in each transaction so my question is about what would be required for Sunflower-land to migrate...
I have followed several forum threads where the question was about migrating to zkEVM and where a question was...

**"My project is currently deployed to Polygon PoS, what would be the effort to migrate to Polygon zkEVM?"**

Answer

**"If you are not using any of the unsupported precompile smart contracts, should be very straightforward. Just deploy the same smart contracts using the same tooling. Be aware of dependencies and addresses of other existing smart contracts."**

Question two,

**"What types of dApps can be deployed on Polygon zkEVM?"**

Reply from polygon,

**"Any dApp that is compatible with EVM can be deployed, except for those that require a specific precompiled that is currently not supported by zkEVM.**

Would it be possible for sunflower-land to migrate and what could be done in the background and how much can open source developers help to make such a smooth transition that is minimally noticeable to users?

Have you in the team thought about this and whether it could be a way to go?
I'm not a developer myself so I'm asking you the question and I could think it would be a cool development to work with and maybe extra cool for all the open source developers who already help with so much and who have an equally big passion for SFL that I have where we can see that there is a bright future for SFL especially when the crypto market turns for the better and users join…

Gas fees have become such a big issue that they are taking up more and more space in Discord and are becoming more and more annoying for users which eventually lead to them giving up and moving on...
I know you have different ideas to make it better, but instead of making small adjustments as you go, maybe it would be better to take the step to migrate to zkEVM if it can now be done smoothly and easily...

I'm a little curious how you think about it and what would be possible so that it would be best for everyone??
The boring environment there becomes in Discord because of gas fees takes away so much from the nice environment there has been before and it's a bit sad even I spend less time there because it's constantly a negative tone as soon as gas fees get higher and it becoming more and more frequent....

**Comment by adamhannigan** [comment]
Hey @LandLadyFarmer Thanks for raising the discussion.

Right now the team are exploring a range of options including:
- Gas price optimisations
- ZkEVM
- Supernets

The later 2 require significant amount of work and are relatively new. We are playing around with these technologies and it is likely apart of our future. At the same time we don't want to be the earliest adopters and the guinea pigs while these systems are still improving.

Supernets is a topic that our CTO is deeply interested in - it is a major change but also very promising being able to use our native SFL as the gas token.

We 100% agree on the gas issues and are actively trying to solve it. Any more feedback/ideas are much appreciated 🙏
